homeassistant/components/withings/common.py

```diff
@@ -43,33 +83,316 @@ class ServiceError(HomeAssistantError):
     """Raise when the service has an error."""
 
 
-class ThrottleData:
-    """Throttle data."""
-
-    def __init__(self, interval: int, data: Any):
-        """Initialize throttle data."""
-        self._time = int(time.time())
-        self._interval = interval
-        self._data = data
-
-    @property
-    def time(self) -> int:
-        """Get time created."""
-        return self._time
-
-    @property
-    def interval(self) -> int:
-        """Get interval."""
-        return self._interval
-
-    @property
-    def data(self) -> Any:
-        """Get data."""
-        return self._data
-
-    def is_expired(self) -> bool:
-        """Is this data expired."""
-        return int(time.time()) - self.time > self.interval
+class WithingsAttribute(NamedTuple):
+    """Immutable class for describing withings sensor data."""
+
+    measurement: Measurement
+    measute_type: Enum
+    friendly_name: str
+    unit_of_measurement: str
+    icon: Optional[str]
+    platform: str
+
+
+class WithingsData(NamedTuple):
+    """Represents value and meta-data from the withings service."""
+
+    attribute: WithingsAttribute
+    value: Any
+
+
+class WebhookConfig(NamedTuple):
+    """Config for a webhook."""
+
+    id: str
+    url: str
+    enabled: bool
+
+
+class StateData(NamedTuple):
+    """State data held by data manager for retrieval by entities."""
+
+    unique_id: str
+    state: Any
+
+
+WITHINGS_ATTRIBUTES = [
+    WithingsAttribute(
+        Measurement.WEIGHT_KG,
+        MeasureType.WEIGHT,
+        "Weight",
+        MASS_KILOGRAMS,
+        "mdi:weight-kilogram",
+        SENSOR_DOMAIN,
+    ),
+    WithingsAttribute(
```

Comment:
I think that Withings should try not to add all possible data by default, but disable some entiteis by deafult that might be less important. The user can then enable them later. This can be controlled by returning `False` from `Entity.entity_registry_enabled_default`. Do you think that the weight/temperature measurement breakdown by type should be disabled by default?

Reply:
This makes sense. Half of the entities do not provide relatively immediate results and can be disabled.
I played around with disabling these by default but I could not get their states to be registered after enabling them. It looks like since the entity is disabled by default, the entity is never added to hass and thus has no place for states. Setting `disabled_by = None` for the `EntityEntry` in the `EntityRegistry` has no effect as it looks like the `async_add_entities` is the only entrypoint for getting entities added to hass.

Reply:
Home Assistant will reload your config entry (unload, then load again) to re-enable entities.

Reply:
What's the preferred method of testing this behavior? Just making reload for the config entry?

Reply:
Patch it so that it returns `True` from `Entity.entity_registry_enabled_default`